File: src/main/java/epicsquid/superiorshields/enchantment/CuringEnchantment.java
package epicsquid.superiorshields.enchantment;

import epicsquid.superiorshields.shield.effect.ShieldEffectMilk;
import net.minecraft.world.item.enchantment.EnchantmentCategory;

public class CuringEnchantment extends ShieldEffectEnchantment<ShieldEffectMilk> {

    public CuringEnchantment(Rarity rarityIn, EnchantmentCategory typeIn) {
        super(rarityIn, typeIn, new ShieldEffectMilk());
    }

    @Override
    public int getMinCost(int enchantmentLevel) {
        return 1;
    }

    @Override
    public int getMaxCost(int enchantmentLevel) {
        return super.getMinCost(enchantmentLevel) + 40;
    }

    @Override
    public int getMaxLevel() {
        return 1;
    }
}

File: src/main/kotlin/dev/epicsquid/superiorshields/enchantment/CuringEnchantment.kt
package dev.epicsquid.superiorshields.enchantment

import dev.epicsquid.superiorshields.effects.EffectHandler
import net.minecraft.world.item.enchantment.EnchantmentCategory

class CuringEnchantment(
	rarity: Rarity,
	category: EnchantmentCategory,
	effectHandler: EffectHandler
) : SuperiorShieldEnchantment(rarity, category, effectHandler) {

	override fun getMaxLevel(): Int = 1

	override fun getMinCost(level: Int): Int = 1

	override fun getMaxCost(level: Int): Int = super.getMinCost(level) + 40

}